### DISCUSSION TITLE
Xamarin Forms upgrading - how to resolve Interfaces

### DISCUSSION THREAD

**Discussion by bb281** [shinyorg/shiny#1450] [Q&A] [answered]
I'm on Xamarin Forms and in the process of upgrading to Shiny 3.x.  Previously, I resolved interfaces by 

`ShinyHost.Resolve<IGpsManager>`

Does anyone know how to resolve this in Xamarin Forms with the new packages?

**Comment by aritchie** [comment, marked as answer]
```csharp
var manager = global::Shiny.Hosting.Host.GetService<IGpsManager>();
```

**Reply by bb281** [reply]
thanks for the quick reply
